### PR TITLE
chore(sdk): allow google-auth < 3

### PR DESCRIPTION
**Pull request by busunkim96**
**Description of your changes:**
google-auth recently published a 2.0.0 release which removed support for Python 2.7. google-auth now requires Python >=3.6. No other breaking changes were made. You can see the full list of changes [here](https://github.com/googleapis/google-auth-library-python/commit/560cf1e).

I'm opening this as @kweinmeister mentioned the `google-auth<2` pin is making it difficult to resolve dependencies with some Google AI libraries.

**Motivation:**
google-auth is a dependency of many different libraries that interact with Google APIs. Increasing the time and number of packages with compatible pins on google-auth lowers the chance end developers who use multiple libraries will see dependency conflicts. 

If possible, please also retain compatibility with 1.x.x versions of google-auth until this issue is resolved
googleapis/google-cloud-python#10566, as that will further reduce the likelihood of diamond dependency conflicts.

Googlers, see [this doc](https://docs.google.com/document/d/1euAvUsia_4zf98lNvpwA3K0o2b4y5Xr9xAkyzCnIMzQ/edit) for more information.

**Checklist:**
- [X] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Review by chensun — COMMENTED** [review]
Can you please add an entry in the release note following this format:
https://github.com/kubeflow/pipelines/blob/2910d10cc38f427f5a8b734210d9f835cdd74631/sdk/RELEASE.md?plain=1#L121

File: sdk/python/requirements.in [hunk 1]
```diff
@@ -21,7 +21,7 @@ kfp-server-api>=1.1.2,<2.0.0
 
 # kfp.Client GCP auth
 google-cloud-storage>=1.20.0,<2
-google-auth>=1.6.1,<2
+google-auth>=1.6.1,<3
```
[the remark this answers is not in the document]
Comment (Bobgy) [reply sdk/python/requirements.in:24]: @chensun https://github.com/kubeflow/pipelines/blob/master/hack/update-all-requirements.sh is our latest script to update all requirements.txt

[the remark this answers is not in the document]
Comment (busunkim96) [reply sdk/python/requirements.in:24]: Hi,

I ran `pip-compile --no-emit-index-url requirements.in` and there don't seem to be any changes to the `requirements.txt`.

I also ran https://github.com/kubeflow/pipelines/blob/master/hack/update-all-requirements.sh and it resulted in a lot of small patch/minor bumps in `backend/requirements.txt` and `backend/src/apiserver/visualization/requirements.txt`, but no changes to the `google-auth` line.

Looking more closely at the dependencies it looks like the current `google-api-python-client` doesn't make it possible to use google-auth 2.0.0. I'll push another commit to adjust that dependency.

**Comment by busunkim96** [comment]
I poked around a bit more (the pip-compile output really helped with this!) to figure out what the dependency conflict @kweinmeister was seeing was coming from.

It turns out some of this library's dependencies require `google-api-python-client>1, <2` which in turn are only compatible with `google-auth<2`.

https://github.com/kubeflow/pipelines/blob/228d37ec5c1e6bd50d6c057e91aba5008b3a5b61/requirements.txt#L44
https://github.com/kubeflow/pipelines/blob/228d37ec5c1e6bd50d6c057e91aba5008b3a5b61/requirements.txt#L48

I don't think there's any harm in merging this first, even though some additional changes will be needed for end users to be able to use `google-auth>2`.

* Internal changelist 411830431 for `tfx-bsl`
* https://github.com/tensorflow/tfx/pull/4486 for `tfx`, `ml-pipelines-sdk`
* https://github.com/googleapis/google-api-python-client/issues/1611

**Comment by chensun** [comment]
> I poked around a bit more (the pip-compile output really helped with this!) to figure out what the dependency conflict @kweinmeister was seeing was coming from.
> 
> It turns out some of this library's dependencies require `google-api-python-client>1, <2` which in turn are only compatible with `google-auth<2`.
> 
> https://github.com/kubeflow/pipelines/blob/228d37ec5c1e6bd50d6c057e91aba5008b3a5b61/requirements.txt#L44
> 
> 
> https://github.com/kubeflow/pipelines/blob/228d37ec5c1e6bd50d6c057e91aba5008b3a5b61/requirements.txt#L48
> 
> I don't think there's any harm in merging this first, even though some additional changes will be needed for end users to be able to use `google-auth>2`.
> 
> * Internal changelist 411830431 for `tfx-bsl`
> * [Allow google-api-python-client < 3 tensorflow/tfx#4486](https://github.com/tensorflow/tfx/pull/4486) for `tfx`, `ml-pipelines-sdk`
> * [Release a 1.x version compatible with `google-auth>2`, `google-api-core>2` for python 3 users googleapis/google-api-python-client#1611](https://github.com/googleapis/google-api-python-client/issues/1611)

Actually I'm not sure if we even should have `pipelines/requirements.txt` in the first place. It would be likely out of sync with `sdk/requirements.txt`.
Also post https://github.com/kubeflow/pipelines/pull/6886, KFP SDK doesn't depend on `google-api-python-client` any more.

I agree we can merge this PR.

/lgtm
/approve

**Pull request merged**